packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -229,7 +257,10 @@ export const useUpdate = <RecordType extends RaRecord = any, ErrorType = Error>(
                 });
             }
 
-            if (mutationOptions.onError && !hasCallTimeOnError.current) {
+            if (callTimeOnError.current) {
+                return callTimeOnError.current(error, variables, context);
+            }
+            if (mutationOptions.onError) {
```

Comment:
**question:** why return at line 261? Don't we want to execute both `callTimeOnError.current` and `mutationOptions.onError` if we have both?

Reply:
No we don't. We have to provide our default side effects when calling the hook (declaration time) but users expect that their own side effects override the default even if they provided it at call time. Otherwise they may end up with 2 notify for instance.